lib/eventsource/eventsource.js

```diff
@@ -0,0 +1,493 @@
+   */
+  constructor (url, eventSourceInitDict = {}) {
+    // 1. Let ev be a new EventSource object.
+    super()
+
+    webidl.argumentLengthCheck(arguments, 1, { header: 'EventSource constructor' })
+
+    if (!experimentalWarned) {
+      experimentalWarned = true
+      process.emitWarning('EventSource is experimental, expect them to change at any time.', {
+        code: 'UNDICI-ES'
+      })
+    }
+
+    url = webidl.converters.USVString(url)
+    eventSourceInitDict = webidl.converters.EventSourceInitDict(eventSourceInitDict)
+
+    // 2. Let settings be ev's relevant settings object.
+    // https://html.spec.whatwg.org/multipage/webappapis.html#environment-settings-object
+    this.#settings = {
+      baseUrl: getGlobalOrigin(),
+      get origin () {
+        return this.baseUrl?.href
+      },
+      policyContainer: {
+        referrerPolicy: 'no-referrer'
+      },
+      lastEventId: '',
+      reconnectionTime: defaultReconnectionTime
+    }
+
+    let urlRecord
+
+    try {
+      // 3. Let urlRecord be the result of encoding-parsing a URL given url, relative to settings.
+      urlRecord = new URL(url, this.#settings.origin)
+    } catch (e) {
+      // 4. If urlRecord is failure, then throw a "SyntaxError" DOMException.
+      throw new DOMException(e, 'SyntaxError')
+    }
+
+    // 5. Set ev's url to urlRecord.
+    this.#url = urlRecord.href
+
+    // 6. Let corsAttributeState be Anonymous.
+    let corsAttributeState = ANONYMOUS
+
+    // 7. If the value of eventSourceInitDict's withCredentials member is true,
+    // then set corsAttributeState to Use Credentials and set ev's
+    // withCredentials attribute to true.
+    if (eventSourceInitDict.withCredentials) {
+      corsAttributeState = USE_CREDENTIALS
+      this.#withCredentials = true
+    }
+
+    // 8. Let request be the result of creating a potential-CORS request given
+    // urlRecord, the empty string, and corsAttributeState.
+    const initRequest = {
+      redirect: 'follow',
+      keepalive: true,
+      // @see https://html.spec.whatwg.org/multipage/urls-and-fetching.html#cors-settings-attributes
+      mode: 'cors',
+      credentials: corsAttributeState === 'anonymous'
+        ? 'same-origin'
+        : 'omit',
+      referrer: 'no-referrer'
+    }
+
+    // 9. Set request's client to settings.
+    initRequest.client = this.#settings
+
+    // 10. User agents may set (`Accept`, `text/event-stream`) in request's header list.
+    initRequest.headers = new HeadersList()
+    initRequest.headers.set('accept', 'text/event-stream', true)
+
+    // 11. Set request's cache mode to "no-store".
+    initRequest.cache = 'no-store'
+
+    // 12. Set request's initiator type to "other".
+    initRequest.initiator = 'other'
+
+    initRequest.urlList = [new URL(this.#url)]
+
+    // 13. Set ev's request to request.
+    this.#request = makeRequest(initRequest)
+
+    this.#request.headersList = initRequest.headers
```

Comment:
@KhafraDev 
You remarked that I should remove this. Please test locally to remove this line and then run the example in examples/eventsource.js 

You will see that the connection will break, because the headers are not set properly. 
Can you please investigate what the issue is? I dont think, that it is something I introduced. 

Thank you :)

Reply:
Changed it, my suggestion was a little off

Reply:
great!